server/render/index.js

```diff
@@ -92,7 +98,7 @@ export function serverRender( req, res ) {
 	if (
 		config.isEnabled( 'server-side-rendering' ) &&
 		context.layout &&
-		! context.user &&
+		! context.isLoggedIn &&
```

Comment:
For reference: This is set here: https://github.com/Automattic/wp-calypso/blob/47a486525668a2637236c2329d9b8e4568a086bd/server/isomorphic-routing/index.js#L68